app/src/main/java/org/mozilla/fenix/components/metrics/AppStartupTelemetry.kt

```diff
@@ -45,17 +54,28 @@ class AppStartupTelemetry(private val metrics: MetricController) : LifecycleObse
 
     fun onFenixApplicationOnCreate() {
         wasAppCreateCalledBeforeActivityCreate = true
+        appLaunchTimeMeasurement.onApplicationOnCreate()
     }
 
-    fun onHomeActivityOnCreate(safeIntent: SafeIntent, hasSavedInstanceState: Boolean) {
-        setOnCreateData(safeIntent, hasSavedInstanceState, false)
+    fun onHomeActivityOnCreate(
+        safeIntent: SafeIntent,
+        hasSavedInstanceState: Boolean,
+        homeActivityInitTimeStampNanoSeconds: Long
+    ) {
+        setOnCreateData(safeIntent, hasSavedInstanceState, homeActivityInitTimeStampNanoSeconds, false)
     }
 
-    fun onExternalAppBrowserOnCreate(safeIntent: SafeIntent, hasSavedInstanceState: Boolean) {
-        setOnCreateData(safeIntent, hasSavedInstanceState, true)
+    fun onExternalAppBrowserOnCreate(
+        safeIntent: SafeIntent,
+        hasSavedInstanceState: Boolean,
+        homeActivityInitTimeStampNanoSeconds: Long
+    ) {
+        setOnCreateData(safeIntent, hasSavedInstanceState, homeActivityInitTimeStampNanoSeconds, true)
     }
 
     fun onHomeActivityOnRestart() {
+        appLaunchTimeMeasurement.onHomeActivityOnRestart()
```

Comment:
nit: mention nothing above here because it measures

Reply:
done